src/services/splitApi.ts

```diff
@@ -53,7 +54,12 @@ export function splitApiFactory(
 
     fetchSplitChanges(since: number, noCache?: boolean, till?: number) {
       const url = `${urls.sdk}/splitChanges?since=${since}${till ? '&till=' + till : ''}${filterQueryString || ''}`;
-      return splitHttpClient(url, noCache ? noCacheHeaderOptions : undefined, telemetryTracker.trackHttp(SPLITS));
+      return splitHttpClient(url, noCache ? noCacheHeaderOptions : undefined, telemetryTracker.trackHttp(SPLITS))
+        .catch((err) => {
+          if (err.statusCode === 414)
+            settings.log.error(ERROR_TOO_MANY_SETS);
+          return err;
```

Comment:
Replace with `throw err;`.
You must not handle the rejected promise here, because the code that calls this method will receive a resolved promise when it should get a rejected one.